server/cart.test.js

```diff
@@ -32,39 +33,89 @@ xdescribe('/api/cart:', () => {
   )
 
   describe('returns items in a users cart:', () => {
-    beforeEach(() => cart.addProduct(product))
 
-    it('returns items in cart:', (done) => {
-      app
+    beforeEach(() => server = request.agent(app))
+
+    it('returns items in cart:', async () => {
+      await server
+        .put(`/api/cart/products/add`)
+        .send(product)
+      const res = await server
         .get(`/api/cart/products`)
-        .expect( res => {
-          expect(res.body.products).to.have.length(1)
-        })
+        .expect(200)
+      expect(res.body).to.have.length(1)
+    }
+      // // V regular promise structure V
+      // server
+      //   .put(`/api/cart/products/add`)
+      //   .send(product)
+      //   .then( () =>
+      //     server.get(`/api/cart/products`)
+      //     .expect(200)
+      //   )
+      //   .then(res => expect(res.body).to.have.length(1))
+    )
+  })
+
+  describe('adds items to a users cart:', () => {
+
+    beforeEach(() => server = request.agent(app))
+
+    it('adds one item to the cart:', (done) => {
+      server
+        .put(`/api/cart/products/add`)
+        .send(product)
         .expect(200, done)
     })
 
-    xit('fails if cart doesn\'t exist', (done) => {
-      app
-        .get(`/api/cart/000/products`)
+    it('can add many of the same item to a cart:', async () => {
+      await server
+        .put(`/api/cart/products/add`)
         .send(product)
-        .expect(404, done)
+      await server
+        .put(`/api/cart/products/add`)
+        .send(product)
+      const res = await server
+        .get(`/api/cart/products`)
+        .expect(200)
+      expect(res.body[0].inCart.quantity).to.equal(2)
     })
   })
 
-  describe('adds an item to a users cart:', () => {
+  describe('subtracts items from a users cart', () => {
 
-    it('adds one item to the cart:', (done) => {
-      app
-        .put(`/api/cart/products`)
+    beforeEach(async () => {
+      server = request.agent(app)
+      await server
+        .put(`/api/cart/products/add`)
```

Comment:
These `add` and `sub` routes aren't really RESTful. Again, if you find yourself using anything but nouns in the resource URI, you're probably doing something more like RPC.

One way to do content-agnostic "differential" requests is with a `PATCH` method, instead of `PUT` which acts like find-or-create. `PATCH` lets you say "this resource, `products`, should be modified according to this diff` — and then you send a request body with something like `addQuantity: 1` or `removeQuantity: 1`. Or any other logic.

I don't recommend you work on this right now though, I'd prioritize progress over REST refactoring.